src/components/SectionTable/GradesPopup.js

```diff
@@ -0,0 +1,98 @@
+import React, { useState } from 'react';
+import { Skeleton } from '@material-ui/lab';
+import { withStyles } from '@material-ui/core/styles';
+import { XAxis, YAxis, CartesianGrid, BarChart, Bar, ResponsiveContainer } from 'recharts';
+import CourseInfoButton from './CourseInfoButton';
+import { queryGrades } from '../../helpers';
+
+const styles = {
+    button: {
+        backgroundColor: '#385EB1',
+        color: '#fff',
+    },
+    gpaTitle: {
+        marginTop: '.5rem',
+        textAlign: 'center',
+        fontWeight: 500,
+        fontSize: '1.2rem',
+        marginRight: '4rem',
+        marginLeft: '4rem',
+    },
+    skeleton: {
+        padding: '4px',
+    },
+};
+
+const GradesPopup = ({ deptCode, courseNumber, classes, isMobileScreen }) => {
+    const [loading, setLoading] = useState(true);
+    const [graphTitle, setGraphTitle] = useState(null);
+    const [gradeData, setGradeData] = useState(null);
+
+    const loadGrades = async () => {
+        if (loading === false) {
+            return;
+        }
+
+        try {
+            let courseGrades = await queryGrades(deptCode, courseNumber);
+
+            let data = [];
+            for (const [key, value] of Object.entries(courseGrades)) {
+                // format data for display in chart
+                // key formatting: sum_grade_a_count -> A
+                if (key !== 'average_gpa') {
+                    data.push({ name: key.split('_')[2]?.toUpperCase(), all: value });
+                }
+            }
+
+            setGraphTitle(`Grade Distribution | Average GPA: ${courseGrades.average_gpa.toFixed(2)}`);
+            setGradeData(data);
+            setLoading(false);
+        } catch (e) {
+            console.log(e);
+            setLoading(false);
+            setGraphTitle('Grades are not available for this class.');
+        }
+    };
+
+    const width = isMobileScreen ? 300 : 500;
+    const height = isMobileScreen ? 200 : 300;
+
+    loadGrades();
+
+    if (loading) {
+        return (
+            <div className={classes.skeleton}>
+                <p>
+                    <Skeleton variant="text" animation="wave" height={height} width={width} />
+                </p>
+            </div>
+        );
+    } else {
+        const encodedDept = encodeURIComponent(deptCode);
+
+        return (
+            <div style={{ marginTop: '5px' }}>
+                <div className={classes.gpaTitle}>{graphTitle}</div>
+                {gradeData && (
+                    <ResponsiveContainer width={width} height={height}>
+                        <BarChart data={gradeData}>
+                            <CartesianGrid strokeDasharray="3 3" />
+                            <XAxis dataKey="name" tick={{ fontSize: 12 }} />
+                            <YAxis tick={{ fontSize: 12 }} width={40} />
+                            <Bar dataKey="all" fill="#5182ed" />
+                        </BarChart>
+                    </ResponsiveContainer>
```

Comment:
Axis labels should be white in dark mode. It's hard to see the gray.

Reply:
Thanks for all the work on the refactor! Code looks a lot cleaner.

changed the axis labels to be white for dark mode with ef68c73.

<img width="504" alt="image" src="https://user-images.githubusercontent.com/30458137/173217110-d0f73dd0-6b1d-425e-a1bf-c7f57041541f.png">